test/DynamoCoreWpfTests/ConnectorViewModelTests.cs

```diff
@@ -0,0 +1,265 @@
+            ///Undo 'unpin' pin
+            Model.ExecuteCommand(new UndoRedoCommand(UndoRedoCommand.Operation.Undo));
+            ///Pin count should be one
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 1);
+        }
+        /// <summary>
+        /// Can undo 'delete pin' command.
+        /// </summary>
+        [Test]
+        public void CanUndoDeletePin()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+
+            ///hard-coded position (1) on wire
+            connectorViewModel.PanelX = 352.666666;
+            connectorViewModel.PanelY = 305.333333;
+            ///Pin connector
+            connectorViewModel.PinConnectorCommand.Execute(null);
+
+            ///Request delete pin
+            var firstPin = connectorViewModel.ConnectorPinViewCollection.First();
+            Model.ExecuteCommand(new DeleteModelCommand(firstPin.Model.GUID));
+            ///Pin count should be zero
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 0);
+            ///Undo delete pin
+            Model.ExecuteCommand(new UndoRedoCommand(UndoRedoCommand.Operation.Undo));
+            ///Pin count should be one
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 1);
+        }
+        /// <summary>
+        /// Can undo 'drag pin'.
+        /// </summary>
+        [Test]
+        public void CanUndoDragPin()
+        {
+            Open(@"UI/ConnectorPinTests.dyn");
+            var connectorViewModel = this.ViewModel.CurrentSpaceViewModel.Connectors.First();
+
+            ///hard-coded position on wire
+            connectorViewModel.PanelX = 352.666666;
+            connectorViewModel.PanelY = 305.333333;
+
+            ///add single pin
+            connectorViewModel.PinConnectorCommand.Execute(null);
+            Assert.AreEqual(connectorViewModel.ConnectorPinViewCollection.Count, 1);
+
+            /// add connectorpin to selection
+            var connectorPin = connectorViewModel.Model.ConnectorPinModels.First();
+            DynamoSelection.Instance.Selection.Add(connectorPin);
+
+            ///initial x, y position of pin
+            double initialX = connectorPin.CenterX;
+            double initialY = connectorPin.CenterY;
+
+            var point = new Point2D(initialX, initialY);
+            var operation = DragSelectionCommand.Operation.BeginDrag;
+            var command = new DragSelectionCommand(point, operation);
+            var selected = DynamoSelection.Instance.Selection;
+            ///execute 'begin drag'
+            Model.ExecuteCommand(command);
+
+            operation = DragSelectionCommand.Operation.EndDrag;
+            point.X += 100;
+            point.Y += 100;
+
+            command = new DragSelectionCommand(point, operation);
```

Comment:
might need some comments on this, not sure i understand why we are executing this command twice

Reply:
Will do, we are calling it once for `begindrag` and a second time for `enddrag`.

Reply:
Done.